read.go

```diff
@@ -226,24 +228,24 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 		buf := make([]int, int(pixelsPerFrame)*samplesPerPixel)
 		for pixel := 0; pixel < int(pixelsPerFrame); pixel++ {
 			for value := 0; value < samplesPerPixel; value++ {
+
+				// read into pre-allocated pixel buffer
+				n, err := d.Read(pixelBuf)
+				if err != nil {
+					return nil, bytesRead,
+						fmt.Errorf("could not read uint%d from input: %w", bitsAllocated, err)
+				}
+				if n < bitsAllocated/8 {
+					return nil, bytesRead,
+						fmt.Errorf("not enough bytes in the input to read uint%d", bitsAllocated)
```

Comment:
just a thought: can we create a sentinel error for this (e.g `ErrorIncompleteRead` or something) and wrap that error with the right bitsAllocated info?

Reply:
also, could we maybe add a [test case here](https://github.com/suyashkumar/dicom/blob/33d864f317adcb467fbe67cceae3052d585ee8b2/read_test.go#L139) to trigger this and expect we get this type of error? we would have to update the test to use `errors.Is`. Also, I can send a change for this later too if we prefer.

Reply:
Done